lib/binding_rust/lib.rs

```diff
@@ -1908,6 +1899,46 @@ fn predicate_error(row: usize, message: String) -> QueryError {
     }
 }
 
+impl fmt::Display for IncludedRangesError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Incorrect range by index: {}", self.0)
+    }
+}
+
+impl fmt::Display for LanguageError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "Incompatible language version {}. Expected minimum {}, maximum {}",
+            self.version, MIN_COMPATIBLE_LANGUAGE_VERSION, LANGUAGE_VERSION,
+        )
+    }
+}
+
+impl fmt::Display for QueryError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let msg = match self.kind {
+            QueryErrorKind::Capture => format!("Invalid capture name {}", self.message),
```

Comment:
Could we avoid the intermediate string allocation here by combining this with the `write!` call somehow?

Reply:
I've just slightly adapted trait implementation from here: https://github.com/tree-sitter/tree-sitter/blob/347b3eb24373ccd9458709cb94654e7fe64657ce/cli/src/error.rs#L56
But off course we can do better on a following commit :smile_cat: